### PR TITLE
fix(godeltaprof): handle renamed runtime.pprof_cyclesPerSecond

### PR DESCRIPTION
**Pull request by korniltsev**
This fixes the build on gotip. The tests are still broken and will be fixed in a followup of https://github.com/grafana/pyroscope-go/issues/103

### CODE REVIEW

File: godeltaprof/compat/stub_go23_test.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+func TestRuntimeExpandFinalInlineFrame(t *testing.T) {
+	checkSignature(t, "runtime/pprof",
+		"runtime_expandFinalInlineFrame",
+		"func runtime/pprof.runtime_expandFinalInlineFrame(stk []uintptr) []uintptr")
+	checkSignature(t, "github.com/grafana/pyroscope-go/godeltaprof/internal/pprof",
+		"runtime_expandFinalInlineFrame",
+		"func github.com/grafana/pyroscope-go/godeltaprof/internal/pprof.runtime_expandFinalInlineFrame(stk []uintptr) []uintptr")
+}
+
+func TestRuntimeCyclesPerSecond(t *testing.T) {
```
Comment (simonswine) [inline godeltaprof/compat/stub_go23_test.go:35]: I think you forgot to remove this test:

https://github.com/grafana/pyroscope-go/blob/ffd26151ec3a9274af9dc11ff96e42bcd249a062/godeltaprof/compat/stub_test.go#L25-L29

**Review by simonswine — APPROVED** [review]
LGTM (- that one suggestion)

File: godeltaprof/compat/stub_go23_test.go (hunk 1 above)
Comment (korniltsev) [reply godeltaprof/compat/stub_go23_test.go:35]: Actually I think all of the tests in `stub_tests.go` are duplicated by the version-specific tests. So I removed all tests from  `stub_test.go`.

**Review by simonswine — APPROVED** [review]
LGTM

**Comment by JacobOaks** [comment]
@korniltsev do you think we can get a tagged release for godeltaprof so there is a non-tip version usable with Go 1.23 which is coming out shortly?

**Comment by korniltsev** [comment]
This rename of `runtime.pprof_cyclesPerSecond` was agreed to be a regression in golang 1.23 and was renamed back, both in gotip https://github.com/golang/go/commit/9a8995b8b6a08d5fe01122771f962b36336f8aec#diff-e9b5b306cc1007e3faa0f5027ac5bcabc39f9d3915deca7cbfa9a20324d4363cR951 and godeltaprof https://github.com/grafana/pyroscope-go/commit/181f95a5c7d7b778a76ef7cbd3261ff310f9857d#diff-0281d9755a602179e47d3b8ca616fdf90f4069f91f3bccfbbfeb53cc72d64e55R26.

There were other issues found https://github.com/grafana/pyroscope-go/issues/103 . They were also fixed both in gotip https://github.com/golang/go/commit/87abb4afb63918d3e5ee5d7ebd196386fa22e368 and godeltaprof https://github.com/grafana/pyroscope-go/commit/181f95a5c7d7b778a76ef7cbd3261ff310f9857d 

Do you have any other issues?

I've just tested on `go version go1.23rc2 linux/amd64` and it seem to compile `godeltaprof/v0.1.7` just fine.

**Pull request merged**